src/lib_json/json_reader.cpp

```diff
@@ -603,9 +603,9 @@ bool Reader::decodeDouble(Token& token, Value& decoded) {
   IStringStream is(buffer);
   is.imbue( std::locale::classic() );
   if (!(is >> value)) {
-    if (value == std::numeric_limits<double>::max())
+    if (buffer.length() > 0 && buffer[0] != '-')
```

Comment:
how does this work with non infinite values?

Reply:
okay, I see it now.  `(is >> value)` will return true for normal values.

Reply:
It probably doesn't.  If it doesn't work doing `if (!(is >> value)) {` I am making an assumption it is only an infinity value.  So if it isn't it would treat it as an infinity value.

Hope that helps! :smiley:

Reply:
This makes sense, appears to fix the issue, passes all of JsonCpp's own tests, and the test in CSMediaLibTest.  I can't think of a realistic case where we'd hit `else if (!std::isinf(value))` now, but maybe this is OK?